File: RIthvik/GroupAnagrams.py
class Solution(object):
    def groupAnagrams(self, strs):
        """
        :type strs: List[str]
        :rtype: List[List[str]]
        """
        groups = defaultdict(list)
        for string in strs:
            groups["".join(sorted(string))].append(string)
        return groups.values()

File: Dyuti/group_anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {

        List<List<String>> groups = new ArrayList<List<String>>();
        List<String> sort = new ArrayList<String>();

        for (String str : strs) {
            char[] chars = str.toCharArray();
            Arrays.sort(chars);
            String sorted = new String(chars);
            
            int index = -1;
            for (int i = 0; i < sort.size(); i++) {
                if (sort.get(i).equals(sorted)) {
                    index = i;
                    break;
                }
            }
            
            if (index == -1) {
                sort.add(sorted);
                List<String> group = new ArrayList<>();
                group.add(str);
                groups.add(group);
            } else {
                groups.get(index).add(str);
            }
        }

        return groups;
    }
    
}
